src/io/image_iter_common.h

```diff
@@ -163,6 +165,8 @@ struct ImageRecParserParam : public dmlc::Parameter<ImageRecParserParam> {
         .describe("Virtually partition the data into these many parts.");
     DMLC_DECLARE_FIELD(part_index).set_default(0)
         .describe("The *i*-th virtual partition to be read.");
+    DMLC_DECLARE_FIELD(device_id).set_default(0)
+        .describe("The device id on which to store the data.");
```

Comment:
Could you expand this documentation text? It is not clear what kind of "device" this refers to and why would you want to set it. Especially since the reader produces data for all of the devices, user will probably be confused which one to choose.

Reply:
@ptrendx agree and thanks for your suggestion. How about changing to "The device id used to create CPUPinned context for internal NDArray."?